compiler/utils/arena/src/lib.rs

```diff
@@ -3,5 +3,52 @@
 #![warn(unreachable_pub)]
 #![warn(clippy::semicolon_if_nothing_returned)]
 
-// For now we use a wrapper around generational-arena
-pub use generational_arena::{Arena, Index};
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd, Hash)]
+pub struct Index {
+    pub ix: usize,
```

Comment:
Nitpick, but can we rename this `ix -> index`? I think avoiding abbreviations would be more consistent with the rest of the code base.

Alternatively, I think a tuple struct would fit this simple wrapper use case even better.

Reply:
Yes, `pub struct Index(pub usize);`